### PR TITLE
feat(geom): create Rectangled/f from point (and set to point)

### PR DESCRIPTION
**Pull request by skaldarnar**
- feat(geom): create Rectangled/f from point (and set to point)
- test(geom): test Rectangled/f created from point

Required by Terasology/PolyWorld#35 (tests).

### CODE REVIEW

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java [hunk 1]
```diff
@@ -170,6 +192,31 @@ public Rectangled set(Rectangled source){
...
+    public Rectangled set(double x, double y) {
+        this.minX = x;
+        this.minY = y;
+        this.maxX = x;
+        this.maxY = y;
+        return this;
+    }
+
+    /**
+     * Set this rectangle to be a clone of <code>source</code>.
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java]:
```suggestion
     * Set this rectangle to be a clone of <code>point</code>.
```

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java [hunk 2]
```diff
@@ -170,6 +192,31 @@ public Rectangled set(Rectangled source){
         return this;
     }
 
+    /**
+     * Set this rectangle to be a clone of <code>source</code>.
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java]: I believe this comment needs updating

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java [hunk 3]
```diff
@@ -170,6 +192,31 @@ public Rectangled set(Rectangled source){
...
+        this.minY = y;
+        this.maxX = x;
+        this.maxY = y;
+        return this;
+    }
+
+    /**
+     * Set this rectangle to be a clone of <code>source</code>.
+     *
+     * @param point the coordinate of both minimum and maximum corner
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectangled.java:213]: This should be in line with the other two. Or rather they with this respectively.

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java [hunk 4]
```diff
@@ -114,42 +129,64 @@ public float maxY() {
...
+    public Rectanglef set(Rectanglef source) {
         this.minX = source.minX;
         this.minY = source.minY;
         this.maxX = source.maxX;
         this.maxY = source.maxY;
         return this;
     }
 
+    /**
+     * Set this rectangle to be a clone of <code>source</code>.
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java]: This comment needs updating.

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java [hunk 5]
```diff
@@ -114,42 +129,64 @@ public float maxY() {
...
+    public Rectanglef set(float x, float y) {
+        this.minX = x;
+        this.minY = y;
+        this.maxX = x;
+        this.maxY = y;
+        return this;
+    }
+
+    /**
+     * Set this rectangle to be a clone of <code>source</code>.
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java]:
```suggestion
     * Set this rectangle to be a clone of <code>point</code>.
```

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java [hunk 6]
```diff
@@ -380,11 +407,9 @@ public boolean isValid() {
     /**
      * Compute the rectangle of intersection between <code>this</code> and the given rectangle.
      * <p>
-     * If the two rectangles do not intersect, then {@link Float#NaN} is stored in each component
-     * of <code>dest</code>.
+     * If the two rectangles do not intersect, then {@link Float#NaN} is stored in each component of <code>dest</code>.
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java:410]: There is no `dest` is there?

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java [hunk 7]
```diff
@@ -394,11 +419,9 @@ public Rectanglef intersection(Rectanglef other) {
...
```
Comment (jdrueckert) [inline joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java:422]: There is no `dest` is there?

**Review by jdrueckert — CHANGES_REQUESTED** [review]

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java (hunk 6 above)
Comment (skaldarnar) [reply joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java:410]: auto-formatting :roll_eyes: I will not go over the unrelated docstrings in this PR.

File: joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java (hunk 7 above)
Comment (skaldarnar) [reply joml-geometry/src/main/java/org/terasology/joml/geom/Rectanglef.java:422]: same here (unrelated formatting change).

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**